File: 2-string/crawler-log-folder.js
/**
 * @param {string[]} logs
 * @return {number}
 */
var minOperations = function(logs) {
    let count = 0
    for (let log of logs) {
        if (log === "../") count = count > 0 ? count-1 : count
        else if (log !== "./") count++
    }
    return count
};

File: 2-string/crawler-log-folder.py
"""
Leet code 1598: Crawler Log Folder
Techniques: stack, greedy
Complexity: O(n) where n is the number of logs
Space Complexity: O(1) for the counter
"""


class Solution(object):
    def minOperations(self, logs):
        count = 0
        for log in logs:
            if log == "../":
                count -= 1 if count > 0 else 0
            elif log != "./":
                count += 1
        return count
